<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>560</width>
    <height>603</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>无损化测试用例生成工具</string>
  </property>
  <widget class="QLineEdit" name="lineEdit_ip_addr">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>50</y>
     <width>161</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="inputMask">
    <string>000.000.000.000</string>
   </property>
   <property name="text">
    <string>...</string>
   </property>
   <property name="maxLength">
    <number>15</number>
   </property>
   <property name="frame">
    <bool>true</bool>
   </property>
   <property name="cursorPosition">
    <number>0</number>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
   <property name="clearButtonEnabled">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineEdit_port">
   <property name="geometry">
    <rect>
     <x>440</x>
     <y>50</y>
     <width>81</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="inputMask">
    <string>00000</string>
   </property>
   <property name="text">
    <string>4444</string>
   </property>
   <property name="maxLength">
    <number>5</number>
   </property>
   <property name="cursorPosition">
    <number>0</number>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_generate">
   <property name="geometry">
    <rect>
     <x>440</x>
     <y>370</y>
     <width>100</width>
     <height>51</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
     <pointsize>15</pointsize>
     <bold>false</bold>
    </font>
   </property>
   <property name="text">
    <string>Generate</string>
   </property>
  </widget>
  <widget class="QToolButton" name="toolButton_save_path">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>200</y>
     <width>26</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="text">
    <string>...</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="checkBox_arch">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>290</y>
     <width>181</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Specify Target Architecture:</string>
   </property>
  </widget>
  <widget class="QComboBox" name="comboBox_payload_type">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>120</y>
     <width>531</width>
     <height>51</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="editable">
    <bool>false</bool>
   </property>
   <property name="currentText">
    <string/>
   </property>
   <property name="currentIndex">
    <number>-1</number>
   </property>
   <property name="placeholderText">
    <string>Payload type...</string>
   </property>
   <property name="duplicatesEnabled">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QLabel" name="label_ip_addr">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>50</y>
     <width>131</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Listener IP Address</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_port">
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>50</y>
     <width>91</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Listener Port</string>
   </property>
  </widget>
  <widget class="Line" name="line_2">
   <property name="geometry">
    <rect>
     <x>340</x>
     <y>100</y>
     <width>211</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label_section_ip_addr">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>15</y>
     <width>301</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
     <pointsize>18</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Listener-Side Configuration</string>
   </property>
  </widget>
  <widget class="Line" name="line_1">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>15</y>
     <width>311</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label_section_payload_main">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>100</y>
     <width>321</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
     <pointsize>18</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Payload Main Parameters Configuration</string>
   </property>
  </widget>
  <widget class="QComboBox" name="comboBox_encoder">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>160</y>
     <width>181</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="currentText">
    <string/>
   </property>
   <property name="placeholderText">
    <string>Encoder...</string>
   </property>
  </widget>
  <widget class="QComboBox" name="comboBox_format">
   <property name="geometry">
    <rect>
     <x>340</x>
     <y>160</y>
     <width>201</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="placeholderText">
    <string>Format...</string>
   </property>
  </widget>
  <widget class="QComboBox" name="comboBox_platform">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>160</y>
     <width>151</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="currentText">
    <string/>
   </property>
   <property name="placeholderText">
    <string>Platform...</string>
   </property>
  </widget>
  <widget class="QComboBox" name="comboBox_arch">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>200</x>
     <y>290</y>
     <width>121</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <item>
    <property name="text">
     <string/>
    </property>
   </item>
  </widget>
  <widget class="QLineEdit" name="lineEdit_save_path">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>200</y>
     <width>331</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
     <pointsize>15</pointsize>
     <italic>true</italic>
    </font>
   </property>
   <property name="inputMask">
    <string/>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
   <property name="placeholderText">
    <string>Please Choose the Save Path</string>
   </property>
   <property name="clearButtonEnabled">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QLabel" name="label_save_path">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>200</y>
     <width>161</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Payload File Save Path</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_section_payload_sub">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>250</y>
     <width>361</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
     <pointsize>18</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Payload Addtional Parameters Configuration</string>
   </property>
  </widget>
  <widget class="Line" name="line_3">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>250</y>
     <width>181</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QCheckBox" name="checkBox_smallest">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>320</y>
     <width>171</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Enable Smallest Encode</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="checkBox_iteration">
   <property name="geometry">
    <rect>
     <x>340</x>
     <y>290</y>
     <width>131</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Specify Iterations:</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_iteration">
   <property name="geometry">
    <rect>
     <x>530</x>
     <y>290</y>
     <width>21</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="text">
    <string> </string>
   </property>
  </widget>
  <widget class="Line" name="line_4">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>350</y>
     <width>541</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QComboBox" name="comboBox_encrypt">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>320</y>
     <width>81</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <item>
    <property name="text">
     <string/>
    </property>
   </item>
   <item>
    <property name="text">
     <string>AES256</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>base64</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>RC4</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>XOR</string>
    </property>
   </item>
  </widget>
  <widget class="QCheckBox" name="checkBox_encrypt">
   <property name="geometry">
    <rect>
     <x>190</x>
     <y>320</y>
     <width>181</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Enable Session Encryption</string>
   </property>
  </widget>
  <widget class="QToolButton" name="toolButton_key">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>440</x>
     <y>320</y>
     <width>101</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="text">
    <string>...</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_key">
   <property name="geometry">
    <rect>
     <x>440</x>
     <y>320</y>
     <width>101</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Upload key file</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QTextBrowser" name="textBrowser_output">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>430</y>
     <width>541</width>
     <height>161</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineEdit_file_name">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>380</y>
     <width>161</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
   <property name="placeholderText">
    <string>sample.exe</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="spinBox_count">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>380</y>
     <width>42</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_count">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>380</y>
     <width>41</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="text">
    <string>Count:</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="spinBox_iteration">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>290</y>
     <width>61</width>
     <height>22</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
    </font>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>380</y>
     <width>131</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Helvetica</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Payload File Name</string>
   </property>
  </widget>
  <zorder>line_4</zorder>
  <zorder>lineEdit_ip_addr</zorder>
  <zorder>lineEdit_port</zorder>
  <zorder>pushButton_generate</zorder>
  <zorder>toolButton_save_path</zorder>
  <zorder>checkBox_arch</zorder>
  <zorder>comboBox_payload_type</zorder>
  <zorder>label_ip_addr</zorder>
  <zorder>label_port</zorder>
  <zorder>line_2</zorder>
  <zorder>label_section_ip_addr</zorder>
  <zorder>line_1</zorder>
  <zorder>label_section_payload_main</zorder>
  <zorder>comboBox_encoder</zorder>
  <zorder>comboBox_format</zorder>
  <zorder>comboBox_platform</zorder>
  <zorder>comboBox_arch</zorder>
  <zorder>lineEdit_save_path</zorder>
  <zorder>label_save_path</zorder>
  <zorder>label_section_payload_sub</zorder>
  <zorder>line_3</zorder>
  <zorder>checkBox_smallest</zorder>
  <zorder>checkBox_iteration</zorder>
  <zorder>label_iteration</zorder>
  <zorder>comboBox_encrypt</zorder>
  <zorder>checkBox_encrypt</zorder>
  <zorder>toolButton_key</zorder>
  <zorder>label_key</zorder>
  <zorder>textBrowser_output</zorder>
  <zorder>lineEdit_file_name</zorder>
  <zorder>spinBox_count</zorder>
  <zorder>label_count</zorder>
  <zorder>spinBox_iteration</zorder>
  <zorder>label</zorder>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>checkBox_arch</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBox_arch</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>95</x>
     <y>299</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_encrypt</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBox_encrypt</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>235</x>
     <y>329</y>
    </hint>
    <hint type="destinationlabel">
     <x>360</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_encrypt</sender>
   <signal>toggled(bool)</signal>
   <receiver>toolButton_key</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>235</x>
     <y>329</y>
    </hint>
    <hint type="destinationlabel">
     <x>512</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_iteration</sender>
   <signal>toggled(bool)</signal>
   <receiver>spinBox_iteration</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>365</x>
     <y>299</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
